main.g

```diff
@@ -11,6 +11,31 @@ GAUSS_ClearUp := function( R,X,R_ )
     fi;
 end;
 
+GAUSS_createHeads := function( pivotrows, pivotcols, width )
+    # inputs: list that contains the row numbers of the pivot rows and
+    # list that contains the column numbers of the pivot cols and
+    # width of matrix
+    local result, i, currentPivot;
+
+    if not Length(pivotrows) = Length(pivotcols) then
+        return [];
+    fi;
+
+    currentPivot := 0;
+    result := [];
```

Comment:
You can initialize the list to the correct length by doing
`ListWithIdenticalEntries( width, 0 );`

Reply:
```suggestion
    result := ListWithIdenticalEntries( width, 0 );
```